src/UI/Internal/Filter/View.elm

```diff
@@ -508,3 +452,299 @@ sortingIcon renderConfig attrs iconSize sorting =
+                    )
+                    sorting
+            , sortAscendingMsg = config.sortAscendingMsg
+            , sortDescendingMsg = config.sortDescendingMsg
+            , clearSortingMsg = config.clearSortingMsg
+            , applied = Maybe.andThen Tuple.first sorting
+            }
+
+        rows renderConfig size =
+            List.map (Element.map config.editMsg) <|
+                case filter of
+                    Model.SingleTextFilter { editable } ->
+                        defaultSingleTextFilter renderConfig size config.label editable
+
+                    Model.MultiTextFilter { editable } ->
+                        defaultMultiTextFilter renderConfig size config.label editable
+
+                    Model.SelectFilter list { editable } ->
+                        selectFilterRender renderConfig size config.label editable list
+
+                    Model.SingleDateFilter { editable } ->
+                        singleDateFilterRender renderConfig size config.label editable
+
+                    Model.RangeDateFilter { editable } ->
+                        rangeDateFilterRender renderConfig size config.label editable
+
+                    Model.PeriodDateFilter { editable } ->
+                        periodDateFilterRender renderConfig size config.label editable
+    in
+    { label = config.label
+    , openMsg = config.openMsg
+    , closeMsg = config.closeMsg
+    , open = config.isOpen
+    , width = Element.fill
+    , rowsHeight = Just 160
+    , size = Medium
+    , sorting = Just sortingData
+    , buttons = defaultButtons config.editMsg filter
+    , applied =
+        Maybe.map
+            (\i -> { preview = String.fromInt i, clearMsg = config.editMsg Msg.Clear })
+            (Model.appliedLength filter)
+    , rows = rows
+    }
+
+
+defaultSingleTextFilter :
+    RenderConfig
+    -> FilterSize
+    -> String
+    -> Model.Editable String
+    -> List (Element Msg)
+defaultSingleTextFilter renderConfig size label editable =
+    editable
+        |> Model.editableWithDefault ""
+        |> TextField.singlelineText Msg.EditSingleText label
+        |> TextField.withSize (sizeToElement size)
+        |> TextField.withWidth TextField.widthFull
+        |> TextField.withOnEnterPressed Msg.Apply
+        |> TextField.renderElement renderConfig
+        |> List.singleton
+
+
+defaultMultiTextFilter :
+    RenderConfig
+    -> FilterSize
+    -> String
+    -> Model.Editable (Array String)
+    -> List (Element Msg)
+defaultMultiTextFilter renderConfig size label editableArr =
+    let
+        rowField subIndex line =
+            line
+                |> TextField.singlelineText (Msg.EditMultiText subIndex) label
+                |> TextField.withSize (sizeToElement size)
+                |> TextField.withWidth TextField.widthFull
+                |> TextField.withOnEnterPressed Msg.Apply
+                |> TextField.renderElement renderConfig
```

Comment:
feels like we could've reused this text field setup

Reply:
I tryed it, and it will just reduce 2 lines in the entire file...